lightning/src/chain/keysinterface.rs

```diff
@@ -397,6 +397,11 @@ pub trait KeysInterface {
 	/// this trait to parse the invoice and make sure they're signing what they expect, rather than
 	/// blindly signing the hash.
 	fn sign_invoice(&self, invoice_preimage: Vec<u8>) -> Result<RecoverableSignature, ()>;
+
+	/// Get a secret key for use in encrypting and decrypting inbound payment data.
+	///
+	/// This method must return the same value each time it is called.
+	fn get_inbound_payment_secret(&self) -> SecretKey;
```

Comment:
This should be a `[u8; 32]`, no?

Reply:
Fine to switch it. Is that because it's a little misleading since it gets expanded into multiple secret keys?

Reply:
We can just document that. Its never used as a `SecretKey`, though, so its strange to pass it through secp256k1 objects.

Reply:
Think we could create a new type `struct Ikm(u8; 32)` for this? Could be used in the next PR for `create_fake_last_hop(last_hop_scid; u64, last_hop_secret: Ikm)`

Reply:
What does the `I` stand for? But, generally, sure.

Reply:
Also the method name including `payment_secret` is confusing given that this is not the same as a `payment_secret` which is later used in the invoice.

Reply:
I was thinking `Ikm` = initial key material. Could just name it `KeyMaterial` instead?

Reply:
Ah, right, yes, duh. For users who aren't steeped in cryptography (or those of us with poor memories) IKM seems a bit confusing. `KeyMaterial` or something seems better to me.